contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -1032,7 +1033,19 @@ update_GrantStmt(Node *n, const char *object, const char *obj_schema, const char
 	if (grantee && stmt->grantees)
 	{
 		RoleSpec   *tmp = (RoleSpec *) llast(stmt->grantees);
-
+		char *cur_dbname = get_cur_db_name();
+		char *pub_grantee = "";
+		strcat(cur_dbname, "_");
+		pub_grantee = cur_dbname;
+		strcat(pub_grantee, "public");
```

Comment:
2. You are CONCATENATING to the already allocated memory - having no enough memory left for the new "public" string.
This is memory corruption!